### PR TITLE
[Feature] ability to send file assignments on a future set date

### LINKED ISSUE #675: [Feature] ability to send file assignments on a future set date

### PR DESCRIPTION
**Pull request by JavaLavaMT**
closes #675

close with associatedemployers/igneous#270

image

image

### CODE REVIEW

File: app/controllers/account/employee/index/documents/new.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import addEdit from 'granite/mixins/controller-abstractions/add-edit';
 export default Controller.extend(addEdit, {
   transitionAfterSave: 'account.employee.index.documents',
   transitionWithModel: false,
+  isExpanded:          false,
```
Comment (James1x0) [inline app/controllers/account/employee/index/documents/new.js]:
```suggestion
```

File: app/models/file-assignment.js [hunk 2]
```diff
@@ -10,11 +10,13 @@ export default Model.extend({
...
-  employee:  belongsTo('employee'),
-  file:      belongsTo('file'),
-  followups: hasMany('file'),
+  creator:     belongsTo('employee'),
+  company:     belongsTo('company'),
+  employee:    belongsTo('employee'),
+  file:        belongsTo('file'),
+  followups:   hasMany('file'),
+  effectiveOn: attr('date'), // Placeholder for effective dated changes. This field is only here to pass along to the api
+
```
Comment (James1x0) [inline app/models/file-assignment.js]:
```suggestion
```

File: app/templates/account/document.hbs [hunk 3]
```diff
@@ -62,7 +62,20 @@
...
+      {{ui-checkbox
+        class="toggle"
+        label="Assign for a future Date"
+        checked=isExpanded
+        onChange=(action (toggle "isExpanded" this))}}
+    </div>
   </div>
+
+  {{#if isExpanded}}
+    {{ui-calendar class="field" type="date" inline=true onChange=(action (mut model.effectiveOn))}}
```
Comment (James1x0) [inline app/templates/account/document.hbs]: Break this out into new lines like the checkbox above.

File: app/templates/account/employee/index/documents/new.hbs [hunk 4]
```diff
@@ -50,6 +50,17 @@
...
 
+    <div class="field center aligned text">
+      {{ui-checkbox
+       class="toggle"
+       label="Assign for a future Date"
+       checked=isExpanded
+       onChange=(action (toggle "isExpanded" this))}}
+    </div>
+    {{#if isExpanded}}
+      {{ui-calendar class="field" type="date" inline=true onChange=(action (mut model.effectiveOn))}}
```
Comment (James1x0) [inline app/templates/account/employee/index/documents/new.hbs]: Breaker breaker

File: app/templates/components/list-item/file-assignment.hbs [hunk 5]
```diff
@@ -5,6 +5,15 @@
...
   {{/link-to}}
+  <div class="detail">
+    {{if pastAssignment 'Assigned' 'Pending'}}
+    {{#ember-tooltip
+      side="right"
+      popperContainer="body"
+    }}
+      Document {{if pastAssignment 'was' 'will be'}} assigned on {{time (or assignment.effectiveOn assignment.created)}} by {{assignment.creator.fullName}}
+    {{/ember-tooltip}}
+  </div>
```
Comment (James1x0) [inline app/templates/components/list-item/file-assignment.hbs]: Can you screenshot how this pairs with the additional "readOn" & "signed" icons?

**Comment by JavaLavaMT** [comment]
image


image

File: app/templates/components/list-item/file-assignment.hbs (hunk 5 above)
Comment (JavaLavaMT) [reply app/templates/components/list-item/file-assignment.hbs]: good catch on this!!!

**Pull request merged**